core/state/statedb.go

```diff
@@ -56,8 +56,13 @@ func (n *proofList) Delete(key []byte) error {
 // StateDB structs within the ethereum protocol are used to store anything
 // within the merkle trie. StateDBs take care of caching and storing
 // nested states. It's the general query interface to retrieve:
+//
 // * Contracts
 // * Accounts
+//
+// Once the state is committed, it's not usable anymore. A new state instance
+// must be created with new root and updated database for accessing latest
+// states.
```

Comment:
that's a big problem for me, as this means calling `OpenTrie`, which becomes an expensive operation with verkle. Is there a way that the trie could be reused?

Reply:
In our case, we always re-create a new state-db after processing a block, even now.

Re reusing the trie, it will be slightly ugly. Perhaps we can introduce some cache layer above the trie database which caches the decoded nodes. So that reloading trie node can be cheap?